src/engine/sparqlExpressions/StringExpressions.cpp

```diff
@@ -454,7 +454,9 @@ class ConcatExpression : public detail::VariadicExpression {
 
   // _________________________________________________________________
   ExpressionResult evaluate(EvaluationContext* ctx) const override {
-    using StringVec = VectorWithMemoryLimit<std::string>;
+    using LiteralVec =
+        VectorWithMemoryLimit<ad_utility::triple_component::Literal>;
+    using Literal = ad_utility::triple_component::Literal;
     // We evaluate one child after the other and append the strings from child i
```

Comment:
I haven't looked at all the details yet, but have a question in principle:

Wouldn't it be appropraite to leave it as a `vector<string> ` where we just concatenate the contents, and then have another vector, where we store the `datatypeOrLanguageTag` that the results will have.
In particular find out, what is the correct behavior when you concat different language tags, or `@en` with `xsd:string` etc, to find out what the correct behavior has to be.
Maybe start with writing unit tests for such cases that mimic the conformance tests.